tracing-flame/src/lib.rs

```diff
@@ -0,0 +1,277 @@
+//! A Tracing [Layer][`FlameLayer`] for generating a folded stack trace for generating flamegraphs
+//! and flamecharts with [`inferno`]
+//!
+//! # Overview
+//!
+//! [`tracing`] is a framework for instrumenting Rust programs to collect
+//! scoped, structured, and async-aware diagnostics. This crate provides helpers
+//! for consuming tracing instrumentation for visualization as a
+//! flamegraph/flamechart which can be useful for finding performance
+//! bottlenecks.
+//!
+//! ## Usage
+//!
+//! This crate is meant to be used as a two step process. First, you capture a
+//! textual representation of the spans that are entered and exited with the
+//! [`FlameLayer`] then you feed these into `inferno-flamegraph` to generate the
+//! flamegraph/flamechart image.
+//!
+//! ## Layer Setup
+//!
+//! ```rust
+//! fn setup_global_subscriber() {
+//!     use tracing_flame::FlameLayer;
+//!     use tracing_subscriber::{registry::Registry, prelude::*};
+//!
+//!     let subscriber = Registry::default()
+//!         .with(FlameLayer::write_to_file("./tracing2.folded").unwrap());
+//!
+//!     tracing::subscriber::set_global_default(subscriber).expect("Could not set global default");
+//! }
```

Comment:
Nit can you show and example with this being set up with `fmt::Layer`? We don't have enough of those examples and it could be useful for folks learning/understanding this.